sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -80,6 +80,9 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
    * @since 1.4.0
    */
   def format(source: String): DataFrameWriter[T] = {
+    if (source.toLowerCase == "hive") {
```

Comment:
In `CatalogImpl.createExternalTable`, we check the hive provider without lowercase, can you fix that? thanks!